src/main/java/org/jenkinsci/plugins/gitclient/GitCredentialBindings.java

```diff
@@ -1,20 +1,22 @@
 package org.jenkinsci.plugins.gitclient;
 
 import com.cloudbees.plugins.credentials.common.StandardCredentials;
-import hudson.EnvVars;
 import hudson.FilePath;
 import hudson.model.TaskListener;
+import hudson.plugins.git.GitTool;
+import org.apache.commons.io.FilenameUtils;
 
 import javax.annotation.Nonnull;
-import java.io.File;
 import java.io.IOException;
 
 public interface GitCredentialBindings {
 
+    String GIT_TOOL_NAME = FilenameUtils.removeExtension(GitTool.getDefaultInstallation().getGitExe());
```

Comment:
I am not sure we should assume that build will be using the git default installation. The user might provide a custom git installation and we should always respect the user's choice first, if not found (or jgit/jgitapache) then fall back to the default installation.

Reply:
You could refer to this piece of code used to resolve user's git installation: https://github.com/jenkinsci/git-plugin/blob/8cb44bfbf8ffb7b3bfcd62a2bc09d6efad54bf3f/src/main/java/hudson/plugins/git/util/GitUtils.java#L94

Reply:
Will definitely look into that and make necessary changes.